### PR TITLE
chore: simplify Price in its own component

### PR DESCRIPTION
**Pull request by zzmp**
- Simplifies the price logic on the toolbar (price is derived from execution price)
- Refactors `Price` to its own component so it can be used in both the toolbar and the summary screen
- Incidentally, add "click to reverse basis" functionality to the summary screen

As part of refactoring price logic, also fixes a bug in `formatLocaleNumber` where `fixedDecimals` was not respected.

### CODE REVIEW

**Comment by willhennessy** [comment]
thanks zach. can you explain "(as price may be different in the other direction, and allowing the user to toggle this is misleading)" ?

**Comment by zzmp** [comment]
Considering that this would be divergent from the interface, where you can click to reverse the price quote basis, I restored that functionality, so "(as price may be different in the other direction, and allowing the user to toggle this is misleading)" is no longer relevant.

To simplify the code (and re-use the component), I've adding the "click to reverse basis" functionality to the summary page.

File: src/lib/components/Swap/Price.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import { ThemedText } from 'lib/theme'
+import formatLocaleNumber from 'lib/utils/formatLocaleNumber'
+import { useCallback, useMemo, useState } from 'react'
+import { formatCurrencyAmount, formatPrice } from 'utils/formatCurrencyAmount'
+
+import { TextButton } from '../Button'
+
+interface PriceProps {
+  trade: Trade<Currency, Currency, TradeType>
+  quoteUSDC?: CurrencyAmount<Token>
```
Comment (ianlapham) [inline src/lib/components/Swap/Price.tsx:14]: Could we just add a comment explaining what `quoteUSDC` is? Took me a second to parse everything

Comment (zzmp) [reply src/lib/components/Swap/Price.tsx:14]: Renamed to `outputUSDC` to clarify this.

**Pull request merged**